src/webchat-ui/components/presentational/previous-conversations/helpers.ts

```diff
@@ -0,0 +1,104 @@
+import moment from "moment";
+import { IMessage } from "../../../../common/interfaces/message";
+import getTextFromMessage, { getMessageAttachmentType } from "../../../../webchat/helper/message";
+import { findReverse } from "../../../utils/find-reverse";
+import { IWebchatConfig } from "../../../../common/interfaces/webchat-config";
+import { PrevConversationsState } from "../../../../webchat/store/previous-conversations/previous-conversations-reducer";
+
+export const getRelativeTime = (messages: IMessage[]) => {
+	const lastMessage = messages[messages.length - 1];
+	if (!lastMessage?.timestamp) return "";
+
+	moment.relativeTimeThreshold("d", 6);
+	moment.relativeTimeThreshold("w", 4); // we need to enable week Threshold
+	moment.updateLocale("en", {
+		relativeTime: {
+			past: "%s",
+			s: "Today",
+			m: "Today",
+			mm: "Today",
+			h: "Today",
+			hh: "Today",
+			d: "Yesterday",
+			dd: "%d days ago",
+			w: "1 week ago",
+			ww: "%d weeks ago",
+			M: "1 month ago",
+			MM: "%d months ago",
+			y: "years",
+			yy: "years",
+		},
+	});
+
+	const relativeTime = moment(lastMessage.timestamp);
+
+	if (relativeTime.fromNow() === "years") {
+		return relativeTime.format("MMMM YYYY");
+	}
+
+	return relativeTime.fromNow();
+};
+
+export const getLastMessagePreview = (messages: IMessage[]) => {
+	const lastReadableMessage = findReverse(messages, message => !!getTextFromMessage(message));
+	if (lastReadableMessage) {
+		return getTextFromMessage(lastReadableMessage);
+	}
+
+	// we don't have text messages in the chat
+	// we need to find the last attachment type
+	const lastMessage = messages[messages.length - 1];
+
+	const attachmentType = getMessageAttachmentType(lastMessage);
+
+	// TODO: implement icons here
+	switch (attachmentType) {
+		case "template":
+			return "template...";
+		case "image":
+			return "image...";
+		case "video":
+			return "video...";
+		case "audio":
+			return "audio...";
+		default:
+			return "...";
```

Comment:
I would use some general text such as "Message" for the default. Ellipsis can be confused with "currently typing" UI convention.

Also capitalize all the items: Image, Video, ...